### ISSUE TITLE
Any HTTPS or HTTP request using a WHATWG URL causes a crash

### ISSUE THREAD

**Issue by JonathanWilbur** [serverless/serverless#5847]
# This is a Bug Report

## Description

* What went wrong?

The NodeJS URL API has two URL types: a Legacy URL and the newer WHATWG URL. Any HTTPS or HTTP request using a WHATWG URL causes a crash. The crash looks like this:

```
  Type Error ---------------------------------------------

  Cannot read property 'startsWith' of undefined
```

This issue stems from the use of this package: https://github.com/TooTallNate/node-https-proxy-agent. I wrote up a bug on the package's GitHub page, which is listed with others below. More details can be found in the other bug write-ups.

* What did you expect should have happened?

> Serverless to not crash.

* What was the config you used?

```yaml
service: name

plugins:
  - serverless-python-requirements

provider:
  name: aws
  runtime: nodejs8.10
  region: us-east-1
  timeout: 10
  bug:
    runtime: nodejs8.10
    handler: source/bug.handler
```

* What stacktrace or error message from your provider did you see?

> None. This was invoked locally.

Similar or dependent issues:

* Read this first: https://github.com/TooTallNate/node-agent-base/issues/26.
* Then this: https://github.com/TooTallNate/node-https-proxy-agent/issues/61.
* And optionally, this: https://github.com/nodejs/node/issues/26198.

## Additional Data

* ***Serverless Framework Version you're using***:

```
  Your Environment Information -----------------------------
     OS:                     win32
     Node Version:           11.7.0
     Serverless Version:     1.37.1
```

* ***Operating System***: It says "win32" above, but it is Windows Professional 64-Bit

## Code that can replicate this error:

```javascript
const https = require("https");
const url = require("url");

let uri = new URL("https://nodejs.org/docs/latest-v8.x/api/url.html");
// Uncomment this one and it will work.
// uri = url.parse("https://nodejs.org/docs/latest-v8.x/api/url.html");

module.exports = {
    handler: function (event, context, callback) {
        https.get(uri, (res) => {
            res.on("close", () => { console.info("Connection closed."); });
            // res.on("data", (chunk) => { console.log("."); });
            res.on("end", () => {
                console.log("Fin.");
                callback(null, "Woop!");
            });
            res.on("error", callback);
        });
    }
}
```

**Comment by eahefnawy** [comment]
@JonathanWilbur just make sure I understand this correctly, you're facing this issue when you're invoking locally? but it works in Lambda?!

**Comment by JonathanWilbur** [comment]
@eahefnawy Yes. That's because, for some reason, HTTP and HTTPS requests appear to be proxied through the aforementioned NodeJS proxy module when the function is invoked locally. Obviously, that does not apply when executed in the AWS runtime.
